### PR TITLE
Switch suggest to search in admin

### PR DESCRIPTION
**Pull request by quaxsze**
fix https://github.com/etalab/data.gouv.fr/issues/599
fix https://github.com/etalab/data.gouv.fr/issues/607

### CODE REVIEW

**Comment by maudetes** [comment]
This will also fix https://github.com/etalab/data.gouv.fr/issues/607, right?

**Comment by quaxsze** [comment]
Switch from a Elasticsearch suggestion to a MongoDB search as you type with the query `icontains`.
The new code takes place in APIV2.
Removed the previous suggest code in APIV1.
Switched the calls from the front (the admin panel) to call APIV2 new mongoDB suggests endpoints.

**Review by maudetes — COMMENTED** [review]
Thank you ! A first batch of reviews :)

Also, there are some useless `from udata import search`.

File: udata/core/dataset/apiv2.py [hunk 1]
```diff
@@ -141,6 +155,73 @@ def get(self):
...
+    'title': fields.String(description='The dataset title'),
+    'acronym': fields.String(description='An optional dataset acronym'),
+    'slug': fields.String(
+        description='The dataset permalink string'),
+    'image_url': fields.String(
+        description='The dataset (organization) logo URL'),
+    'page': fields.UrlFor(
+        'datasets.show_redirect', lambda d: {'dataset': d['slug']},
+        description='The web page URL for this dataset', fallback_endpoint='api.dataset')
+})
```
[the remark this answers is not in the document]
Comment (quaxsze) [reply udata/core/dataset/apiv2.py:178]: The score does not exist in mongoDB as the `contains` query is either true or false

File: udata/api/__init__.py [hunk 2]
```diff
@@ -306,6 +306,7 @@ def init_app(app):
     # Load all core APIs
     import udata.core.activity.api  # noqa
     import udata.core.spatial.api  # noqa
```
[the remark this answers is not in the document]
Comment (quaxsze) [reply udata/api/__init__.py:308]: The `api` of tags had only the suggest endpoint, which was moved to `apiv2`. The tags `api` file was then removed.

File: udata/core/organization/apiv2.py [hunk 3]
```diff
@@ -21,3 +28,46 @@ def get(self):
...
+    'id': fields.String(
+        description='The organization identifier', readonly=True),
+    'name': fields.String(description='The organization name', readonly=True),
+    'acronym': fields.String(
+        description='The organization acronym', readonly=True),
+    'slug': fields.String(
+        description='The organization permalink string', readonly=True),
+    'image_url': fields.String(
+        description='The organization logo URL', readonly=True)
+})
```
[the remark this answers is not in the document]
Comment (quaxsze) [reply udata/core/organization/apiv2.py:52]: 👍

File: udata/settings.py [hunk 4]
```diff
@@ -283,6 +283,30 @@ class Defaults(object):
         'rdf', 'ttl', 'n3',
     ]
 
+    ALLOWED_RESOURCES_MIMES = [
```
Comment (maudetes) [inline udata/settings.py:286]: I made a small visualisation on actual mime types used. We can talk about it later, but some of the most used ones are missing from this list.

File: udata/tests/api/test_datasets_api.py [hunk 5]
```diff
@@ -990,113 +990,103 @@ def test_unfollow_dataset(self):
...
 
-        response = self.get(url_for('api.suggest_mime'),
-                            qs={'q': 'test', 'size': '5'})
+        response = self.get(url_for('apiv2.suggest_mime'),
+                            qs={'q': 'js', 'size': '5'})
         self.assert200(response)
 
         self.assertLessEqual(len(response.json), 5)
-        self.assertGreater(len(response.json), 1)
         # Shortest match first.
```
Comment (maudetes) [inline udata/tests/api/test_datasets_api.py:1046]: I don't think that in our case we sort by the shortest though? We're just being lucky here, no?

File: CHANGELOG.md [hunk 6]
```diff
@@ -9,6 +9,11 @@
   - Those endpoints use MongoDB full text search when `q` argument is used.
   - A new API parser was implemented to replace the search one.
   - The previous ElasticSearch endpoints were moved to APIv2 with the following url pattern: `/{object}/search` (ex: `/datasets/search`).
+- :warning: Suggest changes [#2685](https://github.com/opendatateam/udata/pull/2685):
+  - Current suggest implementation moved from an Elasticsearch index to a MongoDB query using the term `contains`.
+  - The new code is part of APIv2.
+  - The previous code in APIv1 was removed.
+  - JS API calls in the admin panel code were changes to reach APIv2 suggest endpoints.
```
Comment (maudetes) [inline CHANGELOG.md:16]: Maybe add a line on entirely removing the user suggest?

**Review by maudetes — COMMENTED** [review]
- [x] I need to check for the geozone keys, to make sure that we can remove them from the suggest.

Since we are keeping a suggest, but using mongo contains, we could do the same for users, right? Not that it's that needed.

File: udata/settings.py [hunk 7]
```diff
@@ -283,6 +283,30 @@ class Defaults(object):
         'rdf', 'ttl', 'n3',
     ]
```
Comment (maudetes) [inline udata/settings.py:284]: Should we add an `other` if the user don't find its extension in admin?

File: udata/core/reuse/apiv2.py [hunk 8]
```diff
@@ -22,3 +28,44 @@ def get(self):
...
+        args = suggest_parser.parse_args()
+        reuses = Reuse.objects(deleted=None, private__ne=True, title__icontains=args['q'])
+        return [
+            {
+                'id': reuse.id,
+                'title': reuse.title,
+                'slug': reuse.slug,
+                'image_url': reuse.image_url,
+            }
+            for reuse in reuses.order_by(DEFAULT_SORTING).limit(args['size'])
```
[the remark this answers is not in the document]
Comment (quaxsze) [reply udata/core/reuse/apiv2.py:70]: Only returns the top results. The page arg is the object's page not the page number.

**Comment by maudetes** [comment]
> * [X]  I need to check for the geozone keys, to make sure that we can remove them from the suggest.

I think the name + code is enough already to search for geozone!

**Review by maudetes — APPROVED** [review]
I think we're all good for this PR! The user experience with mimes, tags and format seems fine.

Going towards our v2 search architecture :tada:

**Pull request merged**